src/improvement-1.js

```diff
@@ -5,28 +5,18 @@
  */
 
 // FUNCTION 1
-export default function countOfRsInArray(ls) {
-  //changed all vars to let
-  // let count = 0;
-  // for (let i = 0; i < ls.length; i++) {
-  //   if (ls[i] == "R") {
-  //     count++;
-  //   }
-  // }
-  //searching for R globally
+
+//function counts 'R' characters in array or string
+export default function countOfR(item) {
+  //searching for R globally in array or string
   const regex = /R/g;
-  if (typeof ls === "string") {
-    const count = ls.match(regex).length;
+  //statement searches for R if the item is of string type
+  if (typeof item === "string") {
+    const count = item.match(regex).length;
     return count;
-  } else if (typeof ls === "object") {
-    const count = ls.join("").match(regex).length;
+    //statement searches for R if the item is of object type
+  } else if (typeof item === "object") {
```

Comment:
Here you are checking for the type of your input if it's equal to an object then you join them and perform the match. this would be attempted for an object literal too or any other object.